pgvectorscale/src/access_method/graph.rs

```diff
@@ -419,17 +442,43 @@ impl<'a> Graph<'a> {
                     debug_assert!(distance_between_candidate_and_existing_neighbor >= 0.0);
 
                     //factor is high if the candidate is closer to an existing neighbor than the point it's being considered for
-                    let factor =
-                        if distance_between_candidate_and_existing_neighbor < 0.0 + f32::EPSILON {
-                            if distance_between_candidate_and_point < 0.0 + f32::EPSILON {
-                                1.0
+                    let factor = if distance_between_candidate_and_existing_neighbor
```

Comment:
NIT: the `prune_neighborgs` functions is fairly big. Maybe you can abstract the factor calculation away  to its own function. That way you can add docs specific to the factor calculation. Also, you can use guard clauses to remove some of the `if`s.

Reply:
I'm going to ignore this for now. I think we may refactor this later anyway.